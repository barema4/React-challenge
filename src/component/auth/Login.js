import React, { Component } from 'react';
import { withRouter, Link } from 'react-router-dom';
import { connect } from 'react-redux';
import classnames from 'classnames';
import PropTypes from 'prop-types';
import { loginUser } from '../../actions/AuthActions';


export class Login extends Component {
  constructor() {
    super();
    this.state = {
      email: '',
      password: '',
      emailError: '',
      passwordError: '',
      notFoundUser: ''
    };
    this.onChange = this.onChange.bind(this);
    this.onSubmit = this.onSubmit.bind(this);
  }
  componentWillReceiveProps(nextProps) {
    if (nextProps.errors) {
      if (nextProps.errors.email) {
        this.setState({ emailError: nextProps.errors.email });
      } else if (nextProps.errors.password) {
        this.setState({ passwordError: nextProps.errors.password });
      } else {
        this.setState({ notFoundUser: nextProps.errors.error });
      }
    } else {
      window.localStorage.setItem('token', loginUser.nextProps.user.token);
    }
  }
  onChange(e) {
    this.setState({ [e.target.name]: e.target.value });
  }
  onSubmit(e) {
    e.preventDefault();
    const userData = {
      email: this.state.email,
      password: this.state.password
    };
    this.props.loginUser(userData, this.props.history);
  }

  render() {
    const { emailError, passwordError, notFoundUser } = this.state;
    return (
      <section id="login" className="flex-grow-1">
        <div className="container" id="registration-container">
          <div className="row">
            <div className="header">
              <h3 className="display-4 text-center">Welcome Fast Food Fast</h3>
              <form noValidate onSubmit={this.onSubmit}>
                <span>{notFoundUser}</span>
                <div className="form-group">
                  <input
                    type="email"
                    className="form-control form-control-lg"
                    placeholder="Email Address"
                    name="email"
                    value={this.state.email}
                    onChange={this.onChange}
                  />
                  {emailError && (
                    <div className="invalid-feedback">{emailError}</div>
                  )}
                </div>
                <div className="form-group">
                  <input
                    type="password"
                    className="form-control form-control-lg"
                    placeholder="Password"
                    name="password"
                    value={this.state.password}
                    onChange={this.onChange}
                  />
                  {passwordError && (
                    <div className="invalid-feedback">{passwordError}</div>
                  )}
                </div>
                <input
                  type="submit"
                  className="btn btn-primary btn-block mt-4 zeus-color"
                />
                
              </form>
              <div>
                <Link to="/register">Please Sign Up here if you donot have account</Link>
                </div>
            </div>
          </div>
        </div>
        
      </section>
      
    );
  }
}
Login.propType = {
  loginUser: PropTypes.func.isRequired,
  auth: PropTypes.object.isRequired,
  errors: PropTypes.object.isRequired
};
const mapStateToProps = state => {
  return {
    auth: state.auth,
    errors: state.errors
  };
};

export default connect(
  mapStateToProps,
  { loginUser }
)(withRouter(Login));
