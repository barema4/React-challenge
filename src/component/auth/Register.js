import React, { Component } from 'react';
import classnames from 'classnames';
import PropTypes from 'prop-types';
import { connect } from 'react-redux';
import { registerUser } from '../../actions/AuthActions';

export class Register extends Component {
  constructor() {
    super();
    this.state = {
      user_name: '',
      email: '',
      password: '',
      errors: {}
    };
    this.onChange = this.onChange.bind(this);
    this.onSubmit = this.onSubmit.bind(this);
  }

  componentWillReceiveProps(nextProps) {

    if (nextProps.errors) {
      this.setState({ errors: nextProps.errors });
    }
    if (nextProps.errors.message == "Account created successfully") {
      window.location.href = '/login';
    }
  }
  onChange(e) {
    this.setState({ [e.target.name]: e.target.value });
  }

  onSubmit(e) {
    e.preventDefault();
    const newUser = {
      user_name: this.state.user_name,
      email: this.state.email,
      password: this.state.password
    };
    this.props.registerUser(newUser),
      this.setState({ user_name: '', email: '', password: '' });
  }

  render() {
    const { errors } = this.state;

    return (
      <section id="register" className="flex-grow-1">
        <div className="container" id="registration-container">
          <div className="row">
            <div className="header">
              <h1>Please signup with fast food fast</h1>
              <form noValidate onSubmit={this.onSubmit} className="form">
                <div className="form-group">
                  <input
                    type="text"
                    className={classnames('form-control form-control-lg', {
                      'is-invalid': errors.user_name
                    })}
                    placeholder="Username"
                    name="user_name"
                    value={this.state.user_name}
                    onChange={this.onChange}
                  />
                  {errors.user_name && (
                    <div className="invalid-feedback">{errors.user_name}</div>
                  )}
                </div>
                <div className="form-group">
                  <input
                    type="email"
                    className={classnames('form-control form-control-lg', {
                      'is-invalid': errors.email
                    })}
                    placeholder="email Address"
                    name="email"
                    value={this.state.email}
                    onChange={this.onChange}
                  />
                  {errors.email && (
                    <div className="invalid-feedback">{errors.email}</div>
                  )}
                </div>
                <div className="form-group">
                  <input
                    type="password"
                    className={classnames('form-control form-control-lg', {
                      'is-invalid': errors.password
                    })}
                    placeholder="Password"
                    name="password"
                    value={this.state.password}
                    onChange={this.onChange}
                  />
                  {errors.password && (
                    <div className="invalid-feedback">{errors.password}</div>
                  )}
                </div>
                <input
                  type="submit"
                  className="btn btn-primary btn-block mt-4 zeus-color"
                />
              </form>
            </div>
          </div>
        </div>
      </section>
    );
  }
}

Register.propType = {
  registerUser: PropTypes.func.isRequired,
  auth: PropTypes.object.isRequired,
  errors: PropTypes.object.isRequired
};
const mapStateToProps = state => {
  return {
    auth: state.auth.isAuthenticated,
    errors: state.errors
  };
};

export default connect(
  mapStateToProps,
  { registerUser }
)(Register);
