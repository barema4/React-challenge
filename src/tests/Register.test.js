import React from 'react';
import { shallow } from 'enzyme';
import { Register } from '../component/auth/Register';

describe('<Register/>', () => {
  const editor = shallow(<Register registerUser={jest.fn()} />);

  it('should render without crashing', () => {
    expect(editor).toMatchSnapshot();
  });
  const props = {
    registerUser: (newUser) => jest.fn(),
    auth: { isAuthenticated: false, user: {} },
    errors: {}
  };

  it('test Register component onchange', () => {
    const wrapper = shallow(<Register {...props} />);
    const instance = wrapper.instance();
    instance.onChange({ target: { name: 'name', value: 'value' } });
    const e = { preventDefault: () => {} };
    instance.onSubmit(e);
  });

  it('tests that the component receives new props', () => {
    const wrapper = shallow(<Register {...props} />);
    const data = {
      errors: {
        
          name: 'invalid username'
        
      }
    };
    wrapper.instance().componentWillReceiveProps(data);
    expect(wrapper.instance().state.errors.name).toEqual('invalid username');
  });
});
