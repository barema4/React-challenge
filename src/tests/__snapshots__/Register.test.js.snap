// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Register/> should render without crashing 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Register
    registerUser={[MockFunction]}
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "checkPropTypes": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": <div
        className="container"
        id="registration-container"
      >
        <div
          className="row"
        >
          <div
            className="header"
          >
            <h1>
              Please signup with fast food fast
            </h1>
            <form
              className="form"
              noValidate={true}
              onSubmit={[Function]}
            >
              <div
                className="form-group"
              >
                <input
                  className="form-control form-control-lg"
                  name="user_name"
                  onChange={[Function]}
                  placeholder="Username"
                  type="text"
                  value=""
                />
              </div>
              <div
                className="form-group"
              >
                <input
                  className="form-control form-control-lg"
                  name="email"
                  onChange={[Function]}
                  placeholder="email Address"
                  type="email"
                  value=""
                />
              </div>
              <div
                className="form-group"
              >
                <input
                  className="form-control form-control-lg"
                  name="password"
                  onChange={[Function]}
                  placeholder="Password"
                  type="password"
                  value=""
                />
              </div>
              <input
                className="btn btn-primary btn-block mt-4 zeus-color"
                type="submit"
              />
            </form>
          </div>
        </div>
      </div>,
      "className": "flex-grow-1",
      "id": "register",
    },
    "ref": null,
    "rendered": Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": <div
          className="row"
        >
          <div
            className="header"
          >
            <h1>
              Please signup with fast food fast
            </h1>
            <form
              className="form"
              noValidate={true}
              onSubmit={[Function]}
            >
              <div
                className="form-group"
              >
                <input
                  className="form-control form-control-lg"
                  name="user_name"
                  onChange={[Function]}
                  placeholder="Username"
                  type="text"
                  value=""
                />
              </div>
              <div
                className="form-group"
              >
                <input
                  className="form-control form-control-lg"
                  name="email"
                  onChange={[Function]}
                  placeholder="email Address"
                  type="email"
                  value=""
                />
              </div>
              <div
                className="form-group"
              >
                <input
                  className="form-control form-control-lg"
                  name="password"
                  onChange={[Function]}
                  placeholder="Password"
                  type="password"
                  value=""
                />
              </div>
              <input
                className="btn btn-primary btn-block mt-4 zeus-color"
                type="submit"
              />
            </form>
          </div>
        </div>,
        "className": "container",
        "id": "registration-container",
      },
      "ref": null,
      "rendered": Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": <div
            className="header"
          >
            <h1>
              Please signup with fast food fast
            </h1>
            <form
              className="form"
              noValidate={true}
              onSubmit={[Function]}
            >
              <div
                className="form-group"
              >
                <input
                  className="form-control form-control-lg"
                  name="user_name"
                  onChange={[Function]}
                  placeholder="Username"
                  type="text"
                  value=""
                />
              </div>
              <div
                className="form-group"
              >
                <input
                  className="form-control form-control-lg"
                  name="email"
                  onChange={[Function]}
                  placeholder="email Address"
                  type="email"
                  value=""
                />
              </div>
              <div
                className="form-group"
              >
                <input
                  className="form-control form-control-lg"
                  name="password"
                  onChange={[Function]}
                  placeholder="Password"
                  type="password"
                  value=""
                />
              </div>
              <input
                className="btn btn-primary btn-block mt-4 zeus-color"
                type="submit"
              />
            </form>
          </div>,
          "className": "row",
        },
        "ref": null,
        "rendered": Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              <h1>
                Please signup with fast food fast
              </h1>,
              <form
                className="form"
                noValidate={true}
                onSubmit={[Function]}
              >
                <div
                  className="form-group"
                >
                  <input
                    className="form-control form-control-lg"
                    name="user_name"
                    onChange={[Function]}
                    placeholder="Username"
                    type="text"
                    value=""
                  />
                </div>
                <div
                  className="form-group"
                >
                  <input
                    className="form-control form-control-lg"
                    name="email"
                    onChange={[Function]}
                    placeholder="email Address"
                    type="email"
                    value=""
                  />
                </div>
                <div
                  className="form-group"
                >
                  <input
                    className="form-control form-control-lg"
                    name="password"
                    onChange={[Function]}
                    placeholder="Password"
                    type="password"
                    value=""
                  />
                </div>
                <input
                  className="btn btn-primary btn-block mt-4 zeus-color"
                  type="submit"
                />
              </form>,
            ],
            "className": "header",
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": "Please signup with fast food fast",
              },
              "ref": null,
              "rendered": "Please signup with fast food fast",
              "type": "h1",
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": Array [
                  <div
                    className="form-group"
                  >
                    <input
                      className="form-control form-control-lg"
                      name="user_name"
                      onChange={[Function]}
                      placeholder="Username"
                      type="text"
                      value=""
                    />
                  </div>,
                  <div
                    className="form-group"
                  >
                    <input
                      className="form-control form-control-lg"
                      name="email"
                      onChange={[Function]}
                      placeholder="email Address"
                      type="email"
                      value=""
                    />
                  </div>,
                  <div
                    className="form-group"
                  >
                    <input
                      className="form-control form-control-lg"
                      name="password"
                      onChange={[Function]}
                      placeholder="Password"
                      type="password"
                      value=""
                    />
                  </div>,
                  <input
                    className="btn btn-primary btn-block mt-4 zeus-color"
                    type="submit"
                  />,
                ],
                "className": "form",
                "noValidate": true,
                "onSubmit": [Function],
              },
              "ref": null,
              "rendered": Array [
                Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "host",
                  "props": Object {
                    "children": Array [
                      <input
                        className="form-control form-control-lg"
                        name="user_name"
                        onChange={[Function]}
                        placeholder="Username"
                        type="text"
                        value=""
                      />,
                      undefined,
                    ],
                    "className": "form-group",
                  },
                  "ref": null,
                  "rendered": Array [
                    Object {
                      "instance": null,
                      "key": undefined,
                      "nodeType": "host",
                      "props": Object {
                        "className": "form-control form-control-lg",
                        "name": "user_name",
                        "onChange": [Function],
                        "placeholder": "Username",
                        "type": "text",
                        "value": "",
                      },
                      "ref": null,
                      "rendered": null,
                      "type": "input",
                    },
                    undefined,
                  ],
                  "type": "div",
                },
                Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "host",
                  "props": Object {
                    "children": Array [
                      <input
                        className="form-control form-control-lg"
                        name="email"
                        onChange={[Function]}
                        placeholder="email Address"
                        type="email"
                        value=""
                      />,
                      undefined,
                    ],
                    "className": "form-group",
                  },
                  "ref": null,
                  "rendered": Array [
                    Object {
                      "instance": null,
                      "key": undefined,
                      "nodeType": "host",
                      "props": Object {
                        "className": "form-control form-control-lg",
                        "name": "email",
                        "onChange": [Function],
                        "placeholder": "email Address",
                        "type": "email",
                        "value": "",
                      },
                      "ref": null,
                      "rendered": null,
                      "type": "input",
                    },
                    undefined,
                  ],
                  "type": "div",
                },
                Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "host",
                  "props": Object {
                    "children": Array [
                      <input
                        className="form-control form-control-lg"
                        name="password"
                        onChange={[Function]}
                        placeholder="Password"
                        type="password"
                        value=""
                      />,
                      undefined,
                    ],
                    "className": "form-group",
                  },
                  "ref": null,
                  "rendered": Array [
                    Object {
                      "instance": null,
                      "key": undefined,
                      "nodeType": "host",
                      "props": Object {
                        "className": "form-control form-control-lg",
                        "name": "password",
                        "onChange": [Function],
                        "placeholder": "Password",
                        "type": "password",
                        "value": "",
                      },
                      "ref": null,
                      "rendered": null,
                      "type": "input",
                    },
                    undefined,
                  ],
                  "type": "div",
                },
                Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "host",
                  "props": Object {
                    "className": "btn btn-primary btn-block mt-4 zeus-color",
                    "type": "submit",
                  },
                  "ref": null,
                  "rendered": null,
                  "type": "input",
                },
              ],
              "type": "form",
            },
          ],
          "type": "div",
        },
        "type": "div",
      },
      "type": "div",
    },
    "type": "section",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": <div
          className="container"
          id="registration-container"
        >
          <div
            className="row"
          >
            <div
              className="header"
            >
              <h1>
                Please signup with fast food fast
              </h1>
              <form
                className="form"
                noValidate={true}
                onSubmit={[Function]}
              >
                <div
                  className="form-group"
                >
                  <input
                    className="form-control form-control-lg"
                    name="user_name"
                    onChange={[Function]}
                    placeholder="Username"
                    type="text"
                    value=""
                  />
                </div>
                <div
                  className="form-group"
                >
                  <input
                    className="form-control form-control-lg"
                    name="email"
                    onChange={[Function]}
                    placeholder="email Address"
                    type="email"
                    value=""
                  />
                </div>
                <div
                  className="form-group"
                >
                  <input
                    className="form-control form-control-lg"
                    name="password"
                    onChange={[Function]}
                    placeholder="Password"
                    type="password"
                    value=""
                  />
                </div>
                <input
                  className="btn btn-primary btn-block mt-4 zeus-color"
                  type="submit"
                />
              </form>
            </div>
          </div>
        </div>,
        "className": "flex-grow-1",
        "id": "register",
      },
      "ref": null,
      "rendered": Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": <div
            className="row"
          >
            <div
              className="header"
            >
              <h1>
                Please signup with fast food fast
              </h1>
              <form
                className="form"
                noValidate={true}
                onSubmit={[Function]}
              >
                <div
                  className="form-group"
                >
                  <input
                    className="form-control form-control-lg"
                    name="user_name"
                    onChange={[Function]}
                    placeholder="Username"
                    type="text"
                    value=""
                  />
                </div>
                <div
                  className="form-group"
                >
                  <input
                    className="form-control form-control-lg"
                    name="email"
                    onChange={[Function]}
                    placeholder="email Address"
                    type="email"
                    value=""
                  />
                </div>
                <div
                  className="form-group"
                >
                  <input
                    className="form-control form-control-lg"
                    name="password"
                    onChange={[Function]}
                    placeholder="Password"
                    type="password"
                    value=""
                  />
                </div>
                <input
                  className="btn btn-primary btn-block mt-4 zeus-color"
                  type="submit"
                />
              </form>
            </div>
          </div>,
          "className": "container",
          "id": "registration-container",
        },
        "ref": null,
        "rendered": Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": <div
              className="header"
            >
              <h1>
                Please signup with fast food fast
              </h1>
              <form
                className="form"
                noValidate={true}
                onSubmit={[Function]}
              >
                <div
                  className="form-group"
                >
                  <input
                    className="form-control form-control-lg"
                    name="user_name"
                    onChange={[Function]}
                    placeholder="Username"
                    type="text"
                    value=""
                  />
                </div>
                <div
                  className="form-group"
                >
                  <input
                    className="form-control form-control-lg"
                    name="email"
                    onChange={[Function]}
                    placeholder="email Address"
                    type="email"
                    value=""
                  />
                </div>
                <div
                  className="form-group"
                >
                  <input
                    className="form-control form-control-lg"
                    name="password"
                    onChange={[Function]}
                    placeholder="Password"
                    type="password"
                    value=""
                  />
                </div>
                <input
                  className="btn btn-primary btn-block mt-4 zeus-color"
                  type="submit"
                />
              </form>
            </div>,
            "className": "row",
          },
          "ref": null,
          "rendered": Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": Array [
                <h1>
                  Please signup with fast food fast
                </h1>,
                <form
                  className="form"
                  noValidate={true}
                  onSubmit={[Function]}
                >
                  <div
                    className="form-group"
                  >
                    <input
                      className="form-control form-control-lg"
                      name="user_name"
                      onChange={[Function]}
                      placeholder="Username"
                      type="text"
                      value=""
                    />
                  </div>
                  <div
                    className="form-group"
                  >
                    <input
                      className="form-control form-control-lg"
                      name="email"
                      onChange={[Function]}
                      placeholder="email Address"
                      type="email"
                      value=""
                    />
                  </div>
                  <div
                    className="form-group"
                  >
                    <input
                      className="form-control form-control-lg"
                      name="password"
                      onChange={[Function]}
                      placeholder="Password"
                      type="password"
                      value=""
                    />
                  </div>
                  <input
                    className="btn btn-primary btn-block mt-4 zeus-color"
                    type="submit"
                  />
                </form>,
              ],
              "className": "header",
            },
            "ref": null,
            "rendered": Array [
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "children": "Please signup with fast food fast",
                },
                "ref": null,
                "rendered": "Please signup with fast food fast",
                "type": "h1",
              },
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "children": Array [
                    <div
                      className="form-group"
                    >
                      <input
                        className="form-control form-control-lg"
                        name="user_name"
                        onChange={[Function]}
                        placeholder="Username"
                        type="text"
                        value=""
                      />
                    </div>,
                    <div
                      className="form-group"
                    >
                      <input
                        className="form-control form-control-lg"
                        name="email"
                        onChange={[Function]}
                        placeholder="email Address"
                        type="email"
                        value=""
                      />
                    </div>,
                    <div
                      className="form-group"
                    >
                      <input
                        className="form-control form-control-lg"
                        name="password"
                        onChange={[Function]}
                        placeholder="Password"
                        type="password"
                        value=""
                      />
                    </div>,
                    <input
                      className="btn btn-primary btn-block mt-4 zeus-color"
                      type="submit"
                    />,
                  ],
                  "className": "form",
                  "noValidate": true,
                  "onSubmit": [Function],
                },
                "ref": null,
                "rendered": Array [
                  Object {
                    "instance": null,
                    "key": undefined,
                    "nodeType": "host",
                    "props": Object {
                      "children": Array [
                        <input
                          className="form-control form-control-lg"
                          name="user_name"
                          onChange={[Function]}
                          placeholder="Username"
                          type="text"
                          value=""
                        />,
                        undefined,
                      ],
                      "className": "form-group",
                    },
                    "ref": null,
                    "rendered": Array [
                      Object {
                        "instance": null,
                        "key": undefined,
                        "nodeType": "host",
                        "props": Object {
                          "className": "form-control form-control-lg",
                          "name": "user_name",
                          "onChange": [Function],
                          "placeholder": "Username",
                          "type": "text",
                          "value": "",
                        },
                        "ref": null,
                        "rendered": null,
                        "type": "input",
                      },
                      undefined,
                    ],
                    "type": "div",
                  },
                  Object {
                    "instance": null,
                    "key": undefined,
                    "nodeType": "host",
                    "props": Object {
                      "children": Array [
                        <input
                          className="form-control form-control-lg"
                          name="email"
                          onChange={[Function]}
                          placeholder="email Address"
                          type="email"
                          value=""
                        />,
                        undefined,
                      ],
                      "className": "form-group",
                    },
                    "ref": null,
                    "rendered": Array [
                      Object {
                        "instance": null,
                        "key": undefined,
                        "nodeType": "host",
                        "props": Object {
                          "className": "form-control form-control-lg",
                          "name": "email",
                          "onChange": [Function],
                          "placeholder": "email Address",
                          "type": "email",
                          "value": "",
                        },
                        "ref": null,
                        "rendered": null,
                        "type": "input",
                      },
                      undefined,
                    ],
                    "type": "div",
                  },
                  Object {
                    "instance": null,
                    "key": undefined,
                    "nodeType": "host",
                    "props": Object {
                      "children": Array [
                        <input
                          className="form-control form-control-lg"
                          name="password"
                          onChange={[Function]}
                          placeholder="Password"
                          type="password"
                          value=""
                        />,
                        undefined,
                      ],
                      "className": "form-group",
                    },
                    "ref": null,
                    "rendered": Array [
                      Object {
                        "instance": null,
                        "key": undefined,
                        "nodeType": "host",
                        "props": Object {
                          "className": "form-control form-control-lg",
                          "name": "password",
                          "onChange": [Function],
                          "placeholder": "Password",
                          "type": "password",
                          "value": "",
                        },
                        "ref": null,
                        "rendered": null,
                        "type": "input",
                      },
                      undefined,
                    ],
                    "type": "div",
                  },
                  Object {
                    "instance": null,
                    "key": undefined,
                    "nodeType": "host",
                    "props": Object {
                      "className": "btn btn-primary btn-block mt-4 zeus-color",
                      "type": "submit",
                    },
                    "ref": null,
                    "rendered": null,
                    "type": "input",
                  },
                ],
                "type": "form",
              },
            ],
            "type": "div",
          },
          "type": "div",
        },
        "type": "div",
      },
      "type": "section",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "legacyContextMode": "parent",
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getChildContext": Object {
            "calledByRenderer": false,
          },
          "getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
  },
}
`;
